### PR TITLE
[DENA-512] add cockroachdb stock alerts

### PR DESCRIPTION
**Pull request by MarcinGinszt**
I wrote the queries myself, since [example cockroachdb alerts](https://github.com/cockroachdb/cockroach/blob/master/monitoring/rules/alerts.rules.yml) are mostly covered by our container stock alerts.

### CODE REVIEW

File: common/stock/cockroachdb.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,64 @@
+groups:
+  - name: CockroachDB
+    rules:
+      - alert: CockroachdbQueriesErroring
```
Comment (MarcinGinszt) [inline common/stock/cockroachdb.yaml.tmpl:4]: I'm not sure whether we should alert about that- I see errors are happening for a few teams now. Even though I think there is no reason to accept erroring queries.

Or, should we maybe add some higher treshold?

Comment (sbuliarca) [reply common/stock/cockroachdb.yaml.tmpl:4]: Judging by the description of the metric:
```
This metric is a high-level indicator of workload and application degradation with query failures. Use the [Insights page](https://www.cockroachlabs.com/docs/v24.1/ui-insights-page) to find failed executions with their error code to troubleshoot or use application-level logs, if instrumented, to determine the cause of error.
```
it sounds important. I'd say keep it until somebody complains about it and then revisit

File: common/stock/cockroachdb.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      - alert: CockroachdbRangesUnderreplicated
+        expr: (sum(label_replace(ranges_underreplicated,  "namespace", "$1", "kubernetes_namespace", "(.*)")) by (namespace, app) > 1)  * on (namespace) group_left(team) uw_namespace_oncall_team
+        for: 15m
+        labels:
+          alerttype: stock
+          alertgroup: cockroachdb
+        annotations:
+          summary: "CockroachDB cluster {{$labels.app}} in namespace {{$labels.namespace}} reports that some ranges doesn't have enough replicas"
+          impact: "reduced fault tolerance, increased read/ write latency"
+          action: "See replication dashboard, check if there is there enough nodes in the cluster"
```
Comment (sbuliarca) [inline common/stock/cockroachdb.yaml.tmpl:32]: Maybe we should put the link to the actual dashboard here. WDYT?

File: common/stock/cockroachdb.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      - alert: CockroachdbRangesUnavailable
+        expr: (sum(label_replace(ranges_unavailable,  "namespace", "$1", "kubernetes_namespace", "(.*)")) by (namespace, app) > 1)  * on (namespace) group_left(team) uw_namespace_oncall_team
+        for: 15m
+        labels:
+          alerttype: stock
+          alertgroup: cockroachdb
+        annotations:
+          summary: "CockroachDB cluster {{$labels.app}} in namespace {{$labels.namespace}} cannot access some chunks of data"
+          impact: "Data inaccessibility"
+          action: "See replication dashboard, check if there is there enough nodes in the cluster"
```
Comment (sbuliarca) [inline common/stock/cockroachdb.yaml.tmpl:43]:
```suggestion
          action: "See replication dashboard, check if there are enough nodes in the cluster"
```

File: common/stock/cockroachdb.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          link: "https://www.cockroachlabs.com/docs/v24.1/cluster-setup-troubleshooting#replication-issues"
+      - alert: CockroachdbAdmissionOverload
+        expr: (sum(label_replace(admission_io_overload,  "namespace", "$1", "kubernetes_namespace", "(.*)")) by (namespace, app) > 0.8)  * on (namespace) group_left(team) uw_namespace_oncall_team
```
Comment (sbuliarca) [inline common/stock/cockroachdb.yaml.tmpl:46]: Looking at the description in here: https://www.cockroachlabs.com/docs/stable/essential-metrics-dedicated, I think we should alert if greater than 1:
```
If the value of this metric exceeds 1, then it indicates overload. You can also look at the metrics storage.l0-num-files, storage.l0-sublevels or rocksdb.read-amplification directly. A healthy LSM shape is defined as “read-amp < 20” and “L0-files < 1000”, looking at [cluster settings](https://www.cockroachlabs.com/docs/v24.1/cluster-settings) admission.l0_sub_level_count_overload_threshold and admission.l0_file_count_overload_threshold respectively.
```

Comment (sbuliarca) [reply common/stock/cockroachdb.yaml.tmpl:46]: Actually I see you mention it is "close to overload". In this case it's ok 0.8

**Comment by george-angel** [comment]
@MarcinGinszt Have you seen these dashboards?

- https://grafana.prod.aws.uw.systems/d/crdb-console-replications/crdb-console3a-replication
- https://grafana.prod.aws.uw.systems/d/crdb-console-runtime/crdb-console3a-runtime
- https://grafana.prod.aws.uw.systems/d/crdb-console-sql/crdb-console3a-sql
- https://grafana.prod.aws.uw.systems/d/crdb-console-storage/crdb-console3a-storage

File: common/stock/cockroachdb.yaml.tmpl (hunk 2 above)
Comment (MarcinGinszt) [reply common/stock/cockroachdb.yaml.tmpl:32]: Replication dashboard can be found in Cockroachdb localhost
https://www.cockroachlabs.com/docs/stable/ui-replication-dashboard

**Comment by MarcinGinszt** [comment]
@george-angel sorry for late answer! 

`crdb...` dashboards are example dashboards provided by Cockroachlabs- see [this comment](https://github.com/utilitywarehouse/grafana-dashboards/pull/1935#issuecomment-2150206382). They were created just for sake of creating "our" Grafana dashboard (you can see their [saved versions](https://grafana.prod.aws.uw.systems/d/crdb-console-runtime/crdb-console3a-runtime?orgId=1&editview=versions)).

I created our version of CockroachDB dashboard instead of directly using their dashboards, in order to keep the consistency with the other ["common tools" dashboards](https://github.com/utilitywarehouse/grafana-dashboards/tree/master/common_dashboards/dev-enablement).

They indeed contain some metrics that our dashboard doesn't contain (and vice versa). 

I was planning to remove them. WDYT?

**Comment by george-angel** [comment]
> I was planning to remove them. WDYT?

I would prefer if they stayed, I maintain them and they are useful to debug issues which happens often enough to want to maintain them.

If we you want to introduce a new, separate dashboards - thats fine.

File: common/stock/cockroachdb.yaml.tmpl (hunk 2 above)
Comment (sbuliarca) [reply common/stock/cockroachdb.yaml.tmpl:32]: Oh, I see, you need to access it internally 👍 . Thought it's one of our dashboards.

File: common/stock/cockroachdb.yaml.tmpl (hunk 3 above)
Comment (MarcinGinszt) [reply common/stock/cockroachdb.yaml.tmpl]: [Fixed](https://github.com/utilitywarehouse/system-alerts/pull/89/commits/7762a116ecfeb3e6dbe8ec28cfc418db57f33f24), I also [added link to cockroachdb dashboard](https://github.com/utilitywarehouse/system-alerts/pull/89/commits/eeda992dd4724a1bf3a12ef30a59966bf0debe44)

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**